### ISSUE TITLE
Update Rollershutter to use Cover -> what service type?

### ISSUE THREAD

**Issue by n8henrie** [home-assistant/homebridge-homeassistant#38]
Home Assistant recently updated to use `cover` instead of `rollershutter` or `garagedoor`, both of which as being deprecated in the next few releases: https://github.com/home-assistant/home-assistant/pull/2891/commits

I was thinking of updating the rollershutter accessory accordingly, but [this recent PR](https://github.com/home-assistant/homebridge-homeassistant/pull/34#issuecomment-247719956) brings to light another issue. What HomeKit service type should `cover` be?

I initially wrote the rollershutter accessory to use the HomeKit garage door type, in large part as a workaround for [this issue](https://github.com/home-assistant/homebridge-homeassistant/issues/5) -- essentially, I had my garage door opener implemented as a `switch` named `garage door`, but it wouldn't work with Siri because when you ask Siri to "open the garage door," it looks for a device of *type* garage door, independent of what you've named the switch. With rollershutter implemented as the HK garage door type, I can happily ask Siri whether my garage is open and everything is peachy.

However, the PR above brings up #5 again -- if a hass `cover` were implemented as HK garage door type, others wouldn't be able to use their hass `cover`s as HK `windowcover`s, since homebridge-homeassistant uses the hass type to determine the accessory type.

If there were some kind of metadata or alternate way to specify a non-default type, people could customize to have e.g. a homeassistant switch that homebridge interprets as a garagedoor, or perhaps 2 different homeassistant `cover`s -- one of which ends up as a HK garage door and the other as a HK windowcover. Does that make sense?

So, in absence of that functionality currently, if I update the rollershutter accessory to use hass's new `cover`, I would selfishly prefer it to be a HK garage door on the homebridge side, but thought I should open it up to see if there are better ideas on how to work around this issue.

**Comment by nvella** [comment]
Maybe use the Home Assistant configuration.yaml customize section to include information on whether an entity is a rollershutter or garage door.

```yaml
customize:
  cover.my_shutter:
    homebridge_cover_type: rollershutter
  cover.my_garage_door:
    homebridge_cover_type: garage_door
```

**Comment by robbiet480** [comment]
What @nvella suggested could work very nicely. I also wonder if it wouldn't be better for HA itself to have the type of cover as an attribute.

**Comment by n8henrie** [comment]
👍

**Comment by nvella** [comment]
I'm working on a patch for this now using the customise section in HA config.

**Comment by robbiet480** [comment]
Closed via #55.
